feat(forms): Add explicit selectors for all built-in ControlValueAccessors

mrm1st3r opened the pull request:
## PR Checklist
Please check if your PR fulfills the following requirements:

- [X] The commit message follows our guidelines: https://github.com/angular/angular/blob/main/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

- [ ] Bugfix
- [X] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:


## What is the current behavior?

From the number of built-in ControlValueAccessors, only the `DefaultValueAccessor` can be applied to a component indivudially (using the `ngDefaultControl` attribute selector).
All others can only be created by attaching a FormControl or ngModel.

Issue Number: N/A


## What is the new behavior?

All built-in ControlValueAccessors can be created individually using an attribute selector that follows the same naming pattern as `ngDefaultControl`.

This change allows for UI libraries to encapsulate not only text-inputs but also other kinds of inputs without the need to fully implement the ControlValueAccessor. Instead they can act as a proxy to the built-in ControlValueAccessor.

I have built an example that currently works with text-inputs, this PR will make it work with other input-types as well. https://stackblitz.com/edit/angular-ivy-xblubc?file=src/app/custom-input/custom-input.component.ts

## Does this PR introduce a breaking change?

- [ ] Yes
- [X] No


<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->


## Other information

Totati commented:
Shouldn't the selectors include the `[type="xyz"]`? Without them I expect the `<input type="checkbox"  ngNumberControl>` element to be displayed as a checkbox but have the `NumberValueAccessor` initialized.

mrm1st3r commented:
That's a good point. I kept it similar to how the DefaultValueAccessor is built. That also allows to attach to a [type=checkbox] for example. Personally I'd prefer the chance of mixing up the accessors over making the selector more strict than necessary.

mrm1st3r commented:
Hi @dylhunn,
I see you've been assigned as reviewer. What do you think about the change?

dylhunn commented:
Hello! Sorry I hadn't seen this yet -- I'll have a look today.

dylhunn commented:
> Shouldn't the selectors include the [type="xyz"]? Without them I expect the <input type="checkbox" ngNumberControl> element to be displayed as a checkbox but have the NumberValueAccessor initialized.

I'm inclined to agree with this. Is there any drawback to not making the selectors stricter in this way?

mrm1st3r committed:
feat(forms): Add explicit selectors for all built-in ControlValueAccessors

Add explicit selecros for all built-in ControlValueAccessors so they can be instantiated
without the necessity of binding to a FormControl or ngModel

mrm1st3r commented:
Hi @dylhunn.
I changed the selectors to be more strict. Do you have any other remarks?
